Fixes #110 Built the library from skycoin/libskycoin:develop

m4yk3ldev opened the pull request:
Fixes #110 

Changes:
- Submodule change skycoin/skycoin => skycoin/libskycoin@develop
- Changes in url declarations for skycoin/skycoin => skycoin/libskycoin

Does this change need to mentioned in CHANGELOG.md?

Yes

olemis committed:
Branch 0.25.1.dev based on skycoin/skycoin v0.25.0

stdevMac committed:
[stdevMac] refs #86 Add Python client for skycoin

Commit:
refs #110 Added submodule

Commit:
refs #110 Remove submodule skycoin

Commit:
[swig] refs #110 Changes in url in pyskycoin.i

Commit:
refs #110 Changes in setup.py

Commit:
refs #110 Added submodule skycoin/libskyoin

Commit:
refs #110 Repair error by travis in https://travis-ci.org/skycoin/pyskycoin/jobs/491155254#L453

Commit:
refs #110 Added test in travis.yml

Commit:
refs #110 Changes in makefile in test-ci

Commit:
refs #110 Remove test make test-ci the travis.yml

olemis commented on .gitmodules:
```diff
@@ -1,3 +1,6 @@
+[submodule "libskycoin"]
+	path = gopath/src/github.com/skycoin/libskycoin
+	url = https://github.com/skycoin/libskycoin.git
+[submodule "gopath/src/github.com/skycoin/libskycoin"]
```
add submodule just once

stdevHan replied:
ok, thnx

stdevHan replied:
ok, thnx

olemis commented on Makefile:
```diff
@@ -61,8 +60,7 @@ build-swig: ## Generate Python C module from SWIG interfaces
 	rm -f swig/include/swig.h
-	cp -v gopath/src/github.com/skycoin/skycoin/include/swig.h swig/include/
-	swig -python -w501,505,401,302,509,451 -Iswig/include -I$(INCLUDE_DIR) -outdir ./skycoin/ -o swig/pyskycoin_wrap.c $(LIBSWIG_DIR)/pyskycoin.i
+	swig -python -w501,505,401,302,509,451 -Iswig/include -I$(INCLUDE_DIR) -I$(SKYCOIN_DIR)/include -outdir ./skycoin/ -o swig/pyskycoin_wrap.c $(LIBSWIG_DIR)/pyskycoin.i
```
`INCLUDE_DIR = $SKYCOIN_DIR/include`

olemis commented on .travis.yml:
```diff
@@ -82,7 +82,7 @@ before_script:
 script:
 # Test PySkycoin
-- make test-ci
+- make test
```
as before

Commit:
refs #110 Repair error by the suggestion https://github.com/skycoin/pyskycoin/pull/121/files/e0c47179da7a490db16e9b970b9e04510bc973d2

olemis commented on tests/test_cipher_address.py:
```diff
@@ -1,47 +1,5 @@
 import skycoin
 
-def test_TestDecodeBase58Address():
```
why ? how are them addresses checked for validity ?

olemis commented on tests/test_cipher_address.py:
```diff
@@ -1,47 +1,5 @@
 import skycoin
 
-def test_TestDecodeBase58Address():
```
`skycoin/skycoin` is using an exdternal library for `base58` . Please implement this according to this . This function is essential . Needs testing .

stdevHan replied:
I'll just comment, since 0.25.1 is enabled that function

olemis commented:
Another branch +PR is needed to implement this against `v0.25.1` .

olemis committed:
[ci] refs #110 - Absolute paths built from Makefile dir (rather than pwd)

olemis committed:
Merge remote-tracking branch 'simelo/stdevHan_t110_change_submodule_libskycoin' into stdevHan_t110_change_submodule_libskycoin

Commit:
refs #110 Restore test en cipher_address

Commit:
Merge branch '0.25.dev' of https://github.com/skycoin/pyskycoin into stdevHan_t110_change_submodule_libskycoin

Commit:
refs #110 Change in Makefile build-libc

Commit:
refs #110 Changes in Makefile in define GOPATH_DIR

Commit:
refs #110 Changes in Makefile by definitation in GOPATH_DIR and SKYLIBC_DIR

Commit:
refs #110 modifications in the "test_TestDecodeBase58Address" test by eliminating type of errors

Commit:
refs #110  Update submodule

Commit:
refs #110 Update submodule

stdevMac committed:
[stdevMac] refs #86 Update Makefile

stdevMac committed:
[stdevMac] refs #86 Added Python wrapper for skycoin api v0.25.1

stdevMac committed:
[pyrest] refs #86 Set tox test

stdevMac committed:
[pyrest] refs #86 Set Makefile and .travis for pysky_api

stdevMac committed:
Merge branch '0.25.dev' into stdevMac_t86_generate_python_api

stdevMac committed:
[restsharp] refs #86 Solving requested changes for PR

stdevMac committed:
Merge branch 'stdevMac_t86_generate_python_api' of https://github.com/simelo/pyskycoin into stdevMac_t86_generate_python_api

olemis committed:
Merge pull request #118 from simelo/stdevMac_t86_generate_python_api

[restapi] fixes #86 Add Python client for skycoin

stdevMac committed:
[skyapi] refs #86 Update Makefile to Patch openapi generated client

stdevMac committed:
[skyapi] refs #86 Added target for support specific Skycoin node
Update Changelog
Update Makefile for target for specific node address
Set https://staging.node.skycoin.net as default

stdevMac committed:
[skyapi] refs #86 Update host options

stdevMac committed:
[stdevMac] refs #127 Update Makefile

Commit:
Merge branch '0.25.dev' of https://github.com/skycoin/pyskycoin into stdevHan_t110_change_submodule_libskycoin

Commit:
[Makefile] refs #110 Changes in PYTHON_BIN => PYTHON

Commit:
[submodule] refs #110 Update submodule

Commit:
[Makefile] refs #110 Repair error in definition PYTHON_BIN

Commit:
[Makefile] refs #110 Repair error in PYTHON_BIN   ?= python => PYTHON_BIN   ?= python$(PYTHON)

Commit:
[setup.py] refs #110 Changes in setup.py to build folder

stdevMac committed:
[skyapi] refs #86 Update basic tests

stdevMac committed:
[skyapi] refs #86 Updated skyapi library position in working dir

stdevMac committed:
[skyapi] refs #86 Update install-linux.sh for travis

stdevMac committed:
[skyapi] refs #86 Updated MANIFEST.in for new location of skyapi

stdevMac committed:
[skyapi] Updated pyenv in travis instalation

Commit:
[setup.py] refs #110 Changes in definition make build and  repair MANIFEST.in

Commit:
[Makefile] refs #110 Defining PYTHON_BIN according to the OS

Commit:
[travis] refs # Repair version the PYTHON in ok 2.7

Commit:
[travis] refs #110 Change value in env os 2.7 PYTHON=2.7 => PYTHON=2.7.14

stdevMac committed:
Merge remote-tracking branch 'origin/0.25.dev' into stdevMac_t86_generate_python_api

stdevMac committed:
[skyapi] refs #86 Updated pyenv for travis

stdevMac committed:
[skyapi] refs #86 Updated pyenv in osx installation

Commit:
[travis] refs #110 Rename - PYTHON=2.7.14 => - PYTHON=2.7 in env GNU/Linux

olemis committed:
refs #86 - Single line in CHANGELOG

olemis committed:
[skyapi] refs #86 - Update top-level title in README

olemis committed:
Merge pull request #125 from simelo/stdevMac_t86_generate_python_api

fixes #86 - Generate Python client for Skycoin node REST API

Commit:
Merge branch '0.25.dev' of https://github.com/skycoin/pyskycoin into stdevHan_t110_change_submodule_libskycoin

Commit:
[test] refs #110 Comment test test_TestDecodeBase58Address to missing SKY_base58_Hex2Base58

Commit:
[test][submodule] refs #110 Update submodule and discomment test test_TestDecodeBase58Address

Commit:
refs #110 Update submodule to v0.25dev

stdevMac committed:
[skyapi] refs #86 Updated skyapi for new swagger spec

stdevMac committed:
[skypai] refs #86 Updated skycoin contact

olemis committed:
Merge pull request #137 from simelo/stdevMac_t86_generate_python_api

FIxes #136 Updated skyapi to new swagger spec

Commit:
[swig][tox] refs #110 Exclude ti build gopath

Commit:
Merge branch '0.25.dev' of https://github.com/skycoin/pyskycoin into stdevHan_t110_change_submodule_libskycoin

Commit:
[travis] refs #110 Remove step update submodule

Commit:
refs #110 Restructuring of the Makefile and modification in travis as proof

Commit:
[travis] refs #110 Enable make test-ci

Commit:
[Makefile] refs #110 Restore conf

Commit:
[travis] refs #110 Removing the symbolic link and copying the real one

Commit:
[Makefile] refs #110 Restore file conf in make test-ci

Commit:
[submodule] refs #110 Update submodule

Commit:
[submodule] refs #110 Remove submodule skycoin and added submodule libskycoin.

Commit:
[submodule] refs #110 Change develop => v0.25.dev

Commit:
[swig] refs #110 Change url the skycoin => libskycoin

Commit:
[travis] refs #110 Changes in travis to compatibility libskycoin

Commit:
[MANIFEST] refs #110 Changes to convert skycoin => libskycoin

Commit:
[setup] refs #1110 Update url this repo and skypath

Commit:
[Makefile] refs #110 Changes to get all version python

Commit:
[test-py] refs #110 Changes in test to error https://travis-ci.org/skycoin/pyskycoin/jobs/537165269#L3041

Commit:
[MANIFEST] refs #110 Exclude lib

Commit:
[test] refs #110 Comment all test

Commit:
[test] refs #110 Changing value of auxiliary classes due to a OSX error https://travis-ci.org/skycoin/pyskycoin/jobs/537187584

Commit:
[test-libpy] refs #110 Change value en class aux skyok => 0

Commit:
[travis] refs #110 Try install swig 3 in MACOS

Commit:
[sky] refs #110 Remove bad definition  in init

Commit:
[travis] refs #110 Try install swig3.0.12. in brew

Commit:
[travis] refs #110 Try install swig@3.0.4 in matrix OSX

Commit:
[travis] refs #110 Try brew link

Commit:
[travis] refs #110 Downgrade versionswig in MacOS

Commit:
[travis] refs #110 Force link in swig

Commit:
[travis] refs #110 List folder in SWIG OSX

Commit:
[travis] refs #110 Reload conf bash

Commit:
[travis] refs #110 Reload conf bash in `source ~/.bash_profile`

Commit:
[travis] refs #110 Declare pure export swig

Commit:
[travis] refs #110 Force export PATH

Commit:
[travis] refs #110 Restore conf install-osx

Commit:
[travis] refs #110 Trying to install swig@3.0.12 by changing the installation file using git.

Commit:
[travis] refs #110 Change branch to checking cong swig

Commit:
[travis] refs #110 Change name package a install

Commit:
[travis] refs #110 Restore conf travis

Commit:
Merge branch 'stdevHan_t110_submodule_libskycoin_0.25.x' of https://github.com/simelo/pyskycoin into stdevHan_t110_change_submodule_libskycoin

 Conflicts:
	.gitmodules
	.travis.yml
	MANIFEST.in
	Makefile
	gopath/src/github.com/skycoin/libskycoin
	swig/include/swig.h
	tests/test_testsuite_test.py

Commit:
[test-py] refs #110  Comment test ´test_TestDecodeBase58Address´ to missing fuction ´SKY_base58_Hex2Base58´

Commit:
[Makefile] refs #110 Remove bad declaration in `make develop`
